### PR TITLE
[py-breathe] New version 4.35.0

### PR DESCRIPTION
**Pull request by greenc-FNAL**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-breathe/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-breathe/package.py]:
```suggestion
    depends_on("py-sphinx@4,5.0.1:", type=("build", "run"), when="@4.35:")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-breathe/package.py
Comment (greenc-FNAL) [reply var/spack/repos/builtin/packages/py-breathe/package.py]: The Breathe developers specifically exclude future major versions of Sphinx from compatibility ("<7"). Historically speaking this is justified because there were changes in the Sphinx API from 4 to 5 and from 5 to 6 that broke Breathe (and other packages like `py-sphinx-design`). This restriction prevents a build error when `py-sphinx` gets a new major version and somebody tries to build an existing version of `py-breathe`, and I would respectfully suggest a stet, here.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-breathe/package.py]: Historically this was hard-coded into the setup.py install_requires, but it seems like they removed the upper bound in the most recent version. I haven't followed development of this package very closely, so I'm not sure what the developers would recommend. We can add a fake upper bound if you think it's safer.

**Review by adamjstewart — DISMISSED** [review]
Will wait for reply to above comment before merging

File: var/spack/repos/builtin/packages/py-breathe/package.py
Comment (greenc-FNAL) [reply var/spack/repos/builtin/packages/py-breathe/package.py]: I thought I'd seen the restriction in a different file, but I must have been mixed up with `py-sphinx-design` since I was dealing with them both at the same time. I have therefore committed your suggested change.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
